include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,310 @@
+     *
+     * \details
+     *
+     * In the banded alignment the iteration of the inner columns is split into two phases. The first phase
+     * reuses the unbanded column computation and assumes that the banded score matrix always starts at the beginning
+     * of the matrix. In the second phase the special interface
+     * seqan3::detail::pairwise_alignment_algorithm_banded::compute_band_column is used to compute the banded column.
+     * The current implementation assumes that the first cell of the current score matrix column is always the first
+     * cell to compute in every column. The respective seqan3::detail::score_matrix_single_column is resized to the band
+     * size plus one additional field to cover the end of the band where the end of the column is not reached yet.
+     * This cell will be never written to but only read from, i.e. it represents minus infinity. This allows the
+     * algorithm to reuse the standard unbanded cell computation.
+     *
+     * The coordinate matrix represents the global matrix index and not the local band coordinate. Data structures that
+     * require the coordinate might need to map the global matrix coordinate to their local coordinate.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is not valid for the given sequences or
+     *         std::bad_alloc if too much memory is allocated.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    int32_t compute_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
+    {
+        size_t sequence1_size = std::ranges::distance(sequence1);
+        size_t sequence2_size = std::ranges::distance(sequence2);
+
+        check_valid_band_configuration(sequence1_size, sequence2_size);
+
+        // ---------------------------------------------------------------------
+        // Initialisation phase: allocate memory and initialise first column.
+        // ---------------------------------------------------------------------
+
+        this->reset_optimum(); // Reset the tracker for the new alignment computation.
+
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
+        coordinate_matrix local_index_matrix{};
+
+        size_t const number_of_columns = sequence1_size + 1;
+        size_t const number_of_rows = sequence2_size + 1;
+        size_t const band_size = upper_diagonal - lower_diagonal + 1;
+
+        local_score_matrix.resize(column_index_type{number_of_columns},
+                                  row_index_type{band_size + 1},
+                                  this->lowest_viable_score());
+        local_index_matrix.resize(column_index_type{number_of_columns}, row_index_type{number_of_rows});
+
+        auto alignment_matrix_it = local_score_matrix.begin();
+        auto indexed_matrix_it = local_index_matrix.begin();
+
+        int32_t row_size = std::max(0, -lower_diagonal);
```

Comment:
```suggestion
        int32_t row_size = std::max<int32_t>(0, -lower_diagonal);
        int32_t const column_size = std::max<int32_t>(0, upper_diagonal);
```

Reply:
I think this is not necessary, since lower_diagonal and upper_diagonal are stored as int32_t. Only on weird platforms were int is not a signed integer this would break, wouldn't it? Or were int it less than 32 bits. But maybe to make it explicitly workable for everything.